scripts/config

```diff
@@ -0,0 +1,257 @@
+
+	# is_public
+	old_is_public="$(ynh_app_setting_get $app is_public)"
+	if [ $old_is_public -eq 1 ]
+	then
+		old_is_public=true
+	else
+		old_is_public=false
+	fi
+	is_public="${YNH_CONFIG_MAIN_IS_PUBLIC_IS_PUBLIC:-$old_is_public}"
+
+#=================================================
+# SHOW_CONFIG FUNCTION FOR 'SHOW' COMMAND
+#=================================================
+
+show_config() {
+	# here you are supposed to read some config file/database/other then print the values
+	# echo "YNH_CONFIG_${PANEL_ID}_${SECTION_ID}_${OPTION_ID}=value"
+
+	echo "YNH_CONFIG_MAIN_EXPORT_EXPORT=$export"
+
+	echo "YNH_CONFIG_MAIN_PAD_CONFIGURATION_PAD_CONFIG_NOCOLORS=$pad_config_nocolors"
+	echo "YNH_CONFIG_MAIN_PAD_CONFIGURATION_PAD_CONFIG_SHOWLINENUMBERS=$pad_config_showlinenumbers"
+	echo "YNH_CONFIG_MAIN_PAD_CONFIGURATION_PAD_CONFIG_CHATANDUSERS=$pad_config_chatandusers"
+	echo "YNH_CONFIG_MAIN_PAD_CONFIGURATION_PAD_CONFIG_ALWAYSSHOWCHAT=$pad_config_alwaysshowchat"
+	echo "YNH_CONFIG_MAIN_PAD_CONFIGURATION_PAD_CONFIG_SHOW_MARKDOWN=$pad_config_show_markdown"
+	echo "YNH_CONFIG_MAIN_PAD_CONFIGURATION_PAD_CONFIG_PAGEVIEW=$pad_config_pageview"
+
+	echo "YNH_CONFIG_MAIN_MYPADS_CONFIGURATION_MYPADS=$mypads"
+	echo "YNH_CONFIG_MAIN_MYPADS_CONFIGURATION_USELDAP=$useldap"
+
+	echo "YNH_CONFIG_MAIN_IS_PUBLIC_IS_PUBLIC=$is_public"
+}
+
+#=================================================
+# MODIFY THE CONFIGURATION
+#=================================================
+
+apply_config() {
+	restart_etherpad=0
+
+	# Change configuration if needed
+	# padOptions noColors
+	if [ "$pad_config_nocolors" != "$old_pad_config_nocolors" ]
+	then
+		ynh_replace_string "\(\"noColors\" *: \).*," "\1$pad_config_nocolors," "$config_file"
+		restart_etherpad=1
+	fi
+
+	# padOptions showLineNumbers
+	if [ "$pad_config_showlinenumbers" != "$old_pad_config_showlinenumbers" ]
+	then
+		ynh_replace_string "\(\"showLineNumbers\" *: \).*," "\1$pad_config_showlinenumbers," "$config_file"
+		restart_etherpad=1
+	fi
+
+	# padOptions chatAndUsers
+	if [ "$pad_config_chatandusers" != "$old_pad_config_chatandusers" ]
+	then
+		ynh_replace_string "\(\"chatAndUsers\" *: \).*," "\1$pad_config_chatandusers," "$config_file"
+		restart_etherpad=1
+	fi
+
+	# padOptions alwaysShowChat
+	if [ "$pad_config_alwaysshowchat" != "$old_pad_config_alwaysshowchat" ]
+	then
+		ynh_replace_string "\(\"alwaysShowChat\" *: \).*," "\1$pad_config_alwaysshowchat," "$config_file"
+		restart_etherpad=1
+	fi
+
+	# Plugin option ep_markdown_default
+	if [ "$pad_config_show_markdown" != "$old_pad_config_show_markdown" ]
+	then
+		ynh_replace_string "\(\"ep_markdown_default\" *: \).*," "\1$pad_config_show_markdown," "$config_file"
+		restart_etherpad=1
+	fi
+
+	# Plugin option ep_page_view_default
+	if [ "$pad_config_pageview" != "$old_pad_config_pageview" ]
+	then
+		ynh_replace_string "\(\"ep_page_view_default\" *: \).*," "\1$pad_config_pageview," "$config_file"
+		restart_etherpad=1
+	fi
+
+	# Export
```

Comment:
I don't really understand why you use the "action" and the "config" features for the export. Is it possible to fit this only in one of this feature ?

Reply:
Actions are used here to install or remove both abiword and libre-office.
But, I want to allow admins to have abiword and libre-office installed if they want, and be able to switch between them quickly with the config panel.

Reply:
Also, config-panel will call the action script and the action script will call the config-panel feature.